src/Courses/CourseDetailsTabs/TopicsTab.tsx

```diff
@@ -1,32 +1,29 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import TopicsList from '../TopicsList';
+import { Accordion, Card, Row, Col, Button } from 'react-bootstrap';
+import { useParams } from 'react-router-dom';
+import AxiosRequest from '../../Hooks/AxiosRequest';
 
 interface TopicsTabProps {
 
 }
 
-const mock_topics = [
-    {topic_name: 'addition', topic_id: 1},
-    {topic_name: 'subtraction', topic_id: 2},
-    {topic_name: 'multiplication', topic_id: 3},
-    {topic_name: 'english', topic_id: 4}
-];
-const mock_units = [
-    {name: 'Unit 1', topics: mock_topics, unit_id: 1}, 
-    {name: 'Unit 2', topics: mock_topics, unit_id: 2},
-    {name: 'Unit 3', topics: mock_topics, unit_id: 3},
-];
-
 export const TopicsTab: React.FC<TopicsTabProps> = () => {
+    const { courseId } = useParams();
+    const [course, setCourse] = useState<any>(null);
+
+    useEffect(() => {
+        (async ()=>{
+            let course = await AxiosRequest.get(`/curriculum/${courseId}`);
```

Comment:
Is there any reason we didn't separate server integration into it's own files

Reply:
What do you mean, `server integration`? Are you asking why we don't pull out all the API requests into a separate file? We could have created named functions to pull in and pass into useEffect, but I'm not sure there's an easy way to handle state with that.